config/crd/bases/e2etest.grpc.io_loadtests.yaml

```diff
@@ -781,20 +781,11 @@ spec:
                     BigQuery.
                   type: string
               type: object
-            scenarios:
-              description: Scenarios provides a list of configurations for testing.
-              items:
-                description: Scenario references a ConfigMap with the configuration
-                  for the driver and the server clients under test.
-                properties:
-                  name:
-                    description: Name identifies the name of the ConfigMap with the
-                      scenario data.
-                    type: string
-                required:
-                - name
-                type: object
-              type: array
+            scenariosJSON:
+              description: 'ScenariosJSON is string with the contents of a Scenarios
```

Comment:
I think at this point we are only supporting the setup where the top level "scenarios" list has a single item. You should document the current and intended state somewhere. Looks like we might NOT want to support multiple scenarios in a single JSON because the json is passed as-is to the driver? Or maybe that's desirable? Not sure, but the behavior should be documented.

Reply:
The driver expects the flag `--scenarios_file` or `--scenarios_json`. Both of these are the Scenarios message, which supports multiple scenarios. We decided to leave it this way, because we did not want to unmarshal, add and marshal the JSON once to add the appropriate `{"scenarios": [ ... ]}` wrapper.

I felt that continuing to support multiple scenarios in the same containers may actually serve as a benefit for advanced usage and debugging. For example, I may schedule a Java scenario and then run a second. I may also run the second alone. There may be cases were hermetic testing hides issues. If I see a degradation in performance in comparison to running it alone, it suggests there may be some leak.

All of our examples will continue to show 1 scenario per load test, and this is how our CI will run for now. However, we left the `scenariosJSON` plural in the case it may be beneficial later.

Reply:
I agree that continuing to support whatever the driver accepts (=multiple scenarios in a single json) makes sense - I just think it might be useful to document that - users won't understand the details of the implementation and thus they won't know whether they should pass multiple scenarios in a single json or embed multiple scenarios (they woudn't even know what the difference between the two options is). So the docs should say something like "while multiple scenarios in a single JSON is supported, for better isolation it is recommended to specify multiple entries of ScenariosJSON because XYZ".

Reply:
We have modified the internal user guide to reflect this. Changes are not yet merged.